packages/components/src/components/hds/app-side-nav/portal/target.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { inject as service } from '@ember/service';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { DEBUG } from '@glimmer/env';
+import Ember from 'ember';
+
+import type { HdsAppSideNavPortalSignature } from './index';
+
+// import { PortalTargetSignature } from 'ember-stargate/components/portal-target';
+interface PortalTargetSignature {
+  Element: HTMLDivElement;
+  Args: {
+    name: string;
+    multiple?: boolean;
+    onChange?: (count: number) => void;
+  };
+  Blocks: {
+    default: [number];
+  };
+}
+
+import type { Registry as Services } from '@ember/service';
+
+interface HdsAppSideNavPortalTargetSignature {
+  Args: PortalTargetSignature['Args'] & {
+    targetName?: HdsAppSideNavPortalSignature['Args']['targetName'];
+  };
+  Element: HTMLDivElement;
+}
+
+export default class HdsAppSideNavPortalTargetComponent extends Component<HdsAppSideNavPortalTargetSignature> {
+  @service router!: Services['router'];
+
+  @tracked numSubnavs = 0;
+  @tracked lastPanelEl: Element | undefined;
+
+  static get prefersReducedMotionOverride(): boolean {
```

Comment:
Blocking: since Ember.testing is going to be deprecated, can you match what SideNav has now? https://github.com/hashicorp/design-system/blob/8d459d3326a16173674de1ab68d6a694816729e7/packages/components/src/components/hds/side-nav/portal/target.ts#L43